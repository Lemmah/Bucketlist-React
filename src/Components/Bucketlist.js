import React, { Component } from 'react'


class Bucketlist extends Component{

	displayItems(bucketlist){
		console.log(bucketlist);
		this.props.getAllBucketlistItems(bucketlist);
	}

	displayBucketlists(){
		if(!this.props.bucketlists){
			return(
				<ul className="list-group">
				  <li className="list-group-item"></li>
				  <li className="list-group-item">
				  	There are no Bucketlists here.
				  </li>
				  <li className="list-group-item"></li>
				</ul>
			);
		} else {
			return (
				<ul className="list-group"> 
						<strong>
							<center>
								Your Bucketlists
							</center>
						</strong>
					{this.props.bucketlists.map((bucketlist) => {
					   return (
					   	<li onClick={this.displayItems.bind(this, bucketlist)} 
					   			key={bucketlist.id}
					   			className="list-group-item bucketlist"
					   	>{bucketlist.name}</li>);
					})}
				</ul>
			);
		}
	}

	render(){
		return(
			<div>
				{this.displayBucketlists()}
			</div>
		);
	}
}

export default Bucketlist;