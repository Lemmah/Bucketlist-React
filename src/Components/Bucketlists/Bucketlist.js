import React, { Component } from 'react'
import CreateForm from './../Commons/CreateForm'


class Bucketlist extends Component{
	constructor(){
		super();
		this.state = {
			formShow: false,
		}

		this.formClose = this.formClose.bind(this);
		this.formOpen = this.formOpen.bind(this);
		this.createBucketlist = this.createBucketlist.bind(this);
	}

	formClose(){
		this.setState({
			formShow: false,
		})
	}

	formOpen(){
		this.setState({
			formShow: true,
		})
	}

	displayItems(bucketlist){
		this.props.getAllBucketlistItems(bucketlist);
	}

	createBucketlist(name){
		const resourceUrl = "/bucketlists"
		let details = {
			"name": name,
		}
		this.props.createBucketlist(resourceUrl,details);
		this.setState({
			formShow: false,
		});
	}

	displayBucketlists(){
		if(!this.props.bucketlists){
			return(
				<table className="table">
				    <thead>
				      <tr>
				        <th>Your Bucketlists</th>
				        <th>
				        	<button
				        		className="btn btn-success btn-xs"
				        		onClick={this.formOpen}
				        	>
									Create New
									</button>
								</th>
				      </tr>
				    </thead>
				    <tbody>
				      <tr>
				      	<td>No Bucketlists -</td>
				      	<td>
				      		Start by creating some.
				      	</td>
				      </tr>
				    </tbody>
				  </table>
			);
		} else {
			if(this.props.bucketlists.length > 0){
				return (
					<table className="table">
				    <thead>
				      <tr>
				        <th>Your Bucketlists</th>
				        <th>
				        	<button
				        		className="btn btn-success btn-xs"
				        		onClick={this.formOpen}
				        	>
									Create New
									</button>
								</th>
				      </tr>
				    </thead>
				    <tbody>
				      {this.props.bucketlists.map((bucketlist) => {
							   return (
							   	<tr 
							   			style={{ cursor: "pointer" }}
							   			onClick={this.displayItems.bind(this, bucketlist)} 
							   			key={bucketlist.id}
							   			className="bucketlist"
							   	><td>{bucketlist.name}</td><td></td></tr>);
							})}
				    </tbody>
				  </table>
				);
			} else {
				return(
					<ul className="list-group">
					  <strong>
								<center>
									Your Bucketlists
								</center>
							</strong>
						<li className="list-group-item"></li>
					  <li className="list-group-item">
					  	There are no Bucketlists here.
					  </li>
					  <li className="list-group-item"></li>
					</ul>
				);	
			}
		}
	}

	render(){
		return(
			<div>
				{this.displayBucketlists()}
				<CreateForm 
					show={this.state.formShow} 
					onHide={this.formClose}
					onCreateResource={this.createBucketlist}
				/>
			</div>
		);
	}
}

export default Bucketlist;