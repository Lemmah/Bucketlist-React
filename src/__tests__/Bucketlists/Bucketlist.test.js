import React from 'react';
import renderer from 'react-test-renderer';
import Bucketlist from '../../Components/Bucketlists/Bucketlist';

it('Displays Bucketlist without Crashing', () => {
	const rendered = renderer.create(
		<Bucketlist />,
		);
	expect(rendered.toJSON()).toMatchSnapshot();
})