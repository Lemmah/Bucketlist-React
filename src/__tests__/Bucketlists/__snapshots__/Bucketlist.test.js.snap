// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays Bucketlist without Crashing 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Your Bucketlists
        </th>
        <th>
          <button
            className="btn btn-success btn-xs"
            onClick={[Function]}
          >
            Create New
          </button>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          No Bucketlists -
        </td>
        <td>
          Start by creating some.
        </td>
      </tr>
      <tr>
        <td>
          <br />
        </td>
        <td>
          <br />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
